packages/flutter/test/cupertino/switch_test.dart

```diff
@@ -433,6 +434,27 @@ void main() {
     expect(tester.widget<Opacity>(find.byType(Opacity).first).opacity, 0.5);
   });
 
+  testWidgets('Switch is using track color when set', (WidgetTester tester) async {
+    const Color trackColor = Colors.pink;
+
+    await tester.pumpWidget(
+      const Directionality(
+        textDirection: TextDirection.ltr,
+        child: Center(
+          child: CupertinoSwitch(
+            value: false,
+            trackColor: trackColor,
+            dragStartBehavior: DragStartBehavior.down,
+            onChanged: null,
+          ),
+        ),
+      ),
+    );
+
+    expect(find.byType(CupertinoSwitch), findsOneWidget);
+    expect(tester.widget<CupertinoSwitch>(find.byType(CupertinoSwitch).first).trackColor, trackColor);
```

Comment:
```suggestion
    expect(find.byType(CupertinoSwitch), paints..rrect(color: trackColor));
```

Only checking `CupertinoSwitch.trackColor` doesn't verify the render object got the right color. You might need to import `mock_canvas.dart` to use `paints.rect`.

Reply:
Yeah, good catch 👍. Still a newbie in Flutter 😄.